File: php/src/Structural/Bridge/MediumSoda.php
<?


#//MediumSoda.java - one of two classes extending the Abstract

require_once 'Soda.php';

class MediumSoda extends Soda
{

	function __construct()
	{
		$this->setSodaImp();
	}

	function pourSoda ()
	{
		$sodaImp = $this->getSodaImp();
		$retval;
		for($i=0; $i < 2; $i++)
			$retval[]= $sodaImp->pourSodaImp();
		return implode(' ',$retval);
	}
}

File: python/src/Structural/Bridge/MediumSoda.py
# - one of two classes extending the Abstract

from Soda import Soda

class MediumSoda(Soda):

	def __init__(self): 
		self.setSodaImp()

	def pourSoda(self) :
		sodaImp = self.getSodaImp()
		retval = []
		for i in range(2):
			retval.append('...glug...')
			retval.append(sodaImp.pourSodaImp())
		return retval

